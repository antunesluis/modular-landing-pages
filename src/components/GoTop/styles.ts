import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};

    border-radius: 0.4rem;
    transition: all 0.2s ease-in-out;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);

    &:hover {
      transform: translateY(-3px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
      background: ${theme.colors.secondaryColor};
    }

    &:active {
      transform: translateY(0);
      box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
      background: ${theme.colors.primaryColor};
    }

    > svg {
      width: 2.4rem;
      height: 2.4rem;
      transition: transform 0.2s ease-in-out;
    }

    &:hover > svg {
      transform: translateY(-3px);
    }

    &:active > svg {
      transform: translateY(0);
    }
  `}
`;
