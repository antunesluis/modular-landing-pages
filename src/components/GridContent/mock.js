export default {
  title: 'Our Digital Solutions',
  html: `
    <h2>Transform Your Business with Our Solutions</h2>
    <p>
      We are experts in developing <strong>innovative</strong> and <strong>customized</strong> 
      solutions for companies looking to stand out in the digital market. Our team combines technical 
      expertise with strategic vision to deliver exceptional results.
    </p>

    <h3>Why choose our company?</h3>

    <ul>
      <li>
        <strong>Proven Experience</strong> - Over 10 years transforming ideas into 
        successful digital solutions
      </li>
      <li>
        <strong>Cutting-edge Technology</strong> - We use the latest market technologies 
        to ensure performance and security
      </li>
      <li>
        <strong>Dedicated Support</strong> - Team always available to assist 
        in all stages of the project
      </li>
    </ul>

    <blockquote>
      "The partnership with the company was fundamental to our business's digital growth. 
      The team is extremely professional and dedicated."
      <cite>- Maria Silva, CEO of TechCorp</cite>
    </blockquote>

    <h3>Our Services</h3>

    <p>
      We offer a complete range of digital services to boost your business:
    </p>

    <table>
      <thead>
        <tr>
          <th>Service</th>
          <th>Benefits</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Web Development</td>
          <td>Responsive, fast, and SEO-optimized websites</td>
        </tr>
        <tr>
          <td>Digital Marketing</td>
          <td>Customized strategies to increase your online presence</td>
        </tr>
        <tr>
          <td>E-commerce</td>
          <td>Robust and secure platforms for online sales</td>
        </tr>
      </tbody>
    </table>

    <h3>Technologies we use</h3>
    <p>
      We work with the most modern technologies in the market, including 
      <code>React</code>, <code>Node.js</code>, and <code>Python</code>. 
      Our stack is constantly updated to ensure the best solutions.
    </p>

    <h3>Contact us</h3>
    <p>
      Want to know more about how we can help your business? 
      <a href="#contact">Click here</a> to speak with one of our consultants or 
      send an email to <a href="mailto:contact@company.com">contact@company.com</a>
    </p>
  `,
};
