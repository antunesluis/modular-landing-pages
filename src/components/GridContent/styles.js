import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xlarge} 0 ${theme.spacings.xlarge};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    width: 100%;
  `}
`;

export const HtmlContent = styled.div`
  ${({ theme }) => css`
    width: 100%;
    overflow-x: hidden; // Prevent horizontal scroll at component level

    p {
      font-size: ${theme.fonts.sizes.medium};
      line-height: 1.5;
      margin: ${theme.spacings.medium} 0;
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: ${theme.spacings.xlarge} 0 ${theme.spacings.xlarge};
      font-weight: 700;
      font-family: ${theme.fonts.family.secondary};
    }

    // Estilização para links
    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    // Estilização para listas
    ul,
    ol {
      margin: ${theme.spacings.small} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
    }

    ul {
      list-style: disc;
    }

    ol {
      list-style: decimal;
    }

    li {
      text-align: left;
    }

    // Estilização para imagens
    img {
      max-width: 100%;
      height: auto;
    }

    // Estilização para citações
    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      color: ${theme.colors.primaryColor};
      margin: ${theme.spacings.medium} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
      font-style: italic;
    }

    // Estilização para tabelas
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      display: block;
      overflow-x: auto; // Adiciona scroll horizontal apenas na tabela
      -webkit-overflow-scrolling: touch; // Melhora a rolagem em dispositivos touch

      // Remove display: block em telas maiores
      @media (min-width: 768px) {
        display: table;
      }
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 1px solid ${theme.colors.mediumGray};
      text-align: left;
      min-width: 15rem; // Garante uma largura mínima para células
    }

    table th {
      background: ${theme.colors.primaryColor};
      color: ${theme.colors.white};
    }
  `}
`;
