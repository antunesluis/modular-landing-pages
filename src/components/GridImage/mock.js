export default {
  background: false,
  title: 'Project Gallery',
  description:
    'Discover some of the transformative projects we have developed for our clients. Each solution is unique and designed to maximize results.',
  grid: [
    {
      altText: 'E-commerce Platform',
      large: 'https://picsum.photos/360/360?r=1',
      medium: 'https://picsum.photos/360/360?r=1',
      small: 'https://picsum.photos/360/360?r=1',
      thumbnail: 'https://picsum.photos/360/360?r=1',
    },
    {
      altText: 'Banking App Interface',
      srcImg: 'https://picsum.photos/360/360?r=2',
      medium: 'https://picsum.photos/360/360?r=2',
      small: 'https://picsum.photos/360/360?r=2',
      thumbnail: 'https://picsum.photos/360/360?r=2',
    },
    {
      altText: 'Healthcare Dashboard',
      srcImg: 'https://picsum.photos/360/360?r=3',
      medium: 'https://picsum.photos/360/360?r=3',
      small: 'https://picsum.photos/360/360?r=3',
      thumbnail: 'https://picsum.photos/360/360?r=3',
    },
    {
      altText: 'Education Platform',
      srcImg: 'https://picsum.photos/360/360?r=4',
      medium: 'https://picsum.photos/360/360?r=4',
      small: 'https://picsum.photos/360/360?r=4',
      thumbnail: 'https://picsum.photos/360/360?r=4',
    },
    {
      altText: 'Real Estate Solution',
      srcImg: 'https://picsum.photos/360/360?r=5',
      medium: 'https://picsum.photos/360/360?r=5',
      small: 'https://picsum.photos/360/360?r=5',
      thumbnail: 'https://picsum.photos/360/360?r=5',
    },
    {
      altText: 'Logistics System',
      srcImg: 'https://picsum.photos/360/360?r=6',
      medium: 'https://picsum.photos/360/360?r=6',
      small: 'https://picsum.photos/360/360?r=6',
      thumbnail: 'https://picsum.photos/360/360?r=6',
    },
  ],
};
