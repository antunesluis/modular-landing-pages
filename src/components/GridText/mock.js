export default {
  background: true, // Enables background with theme-defined image
  title: 'Our Services',
  description:
    'We offer complete software development solutions, focusing on quality and innovation. Our specialized team is ready to turn your ideas into reality.',
  grid: [
    {
      title: 'Technical Consulting',
      description:
        'We provide specialized consulting in software architecture, technology choices, and development best practices. We help your team make the best technical decisions.',
    },
    {
      title: 'Web Development',
      description:
        'We create modern and responsive web applications using the latest market technologies. Our focus is on React, Next.js, and Node.js, ensuring performance and scalability for your project.',
    },
    {
      title: 'UX/UI Design',
      description:
        'Our design team creates intuitive and attractive interfaces, focusing on user experience. We work with prototyping, usability testing, and complete design systems.',
    },
  ],
};
