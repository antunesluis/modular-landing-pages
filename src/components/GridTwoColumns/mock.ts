export default {
  title: 'Digital Innovation for Your Business',
  text: 'Since 2012, we have been a reference in software development and digital transformation. Our team combines technical expertise with strategic vision to create solutions that drive your business growth.',
  srcImg: 'assets/images/illustration/programador.svg',
};
