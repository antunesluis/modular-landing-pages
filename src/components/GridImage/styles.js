import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;
    position: relative;

    > ${TextComponent} {
      opacity: 0.9;
      max-width: 100%;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} ${theme.spacings.small};

      > ${TextComponent} {
        margin: ${theme.spacings.medium} 0 ${theme.spacings.xhuge};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: grid;
    grid-template-columns: repeat(3, 1fr);

    gap: ${theme.spacings.large};
    margin: ${theme.spacings.large} auto;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.medium};
      margin: ${theme.spacings.medium} auto;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    aspect-ratio: 1; // Mantém proporção quadrada
    overflow: hidden;
    position: relative;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);

    border-radius: ${theme.spacings.small};
    background: ${theme.colors.mediumGray};

    &::before {
      content: '';
      z-index: 1;
      opacity: 0;
      position: absolute;
      inset: 0; // Substitui top: 0, left: 0, width: 100%, height: 100%

      background: rgba(0, 0, 0, 0.3);
      transition: opacity 300ms ease-in-out;
    }

    &:hover {
      &::before {
        opacity: 1;
      }

      img {
        transform: scale(1.1) rotate(3deg);
      }
    }
  `}
`;

export const Image = styled.img`
  ${() => css`
    display: block; // Remove espaço extra abaixo da imagem
    width: 100%;
    height: 100%;
    object-fit: cover;
    vertical-align: middle;
    transition: transform 500ms ease-in-out;
    will-change: transform;
  `}
`;

export const ImageCaption = styled.figcaption`
  ${({ theme }) => css`
    position: absolute;
    opacity: 0;
    bottom: 0;
    right: 0; // Garante que ocupe toda a largura
    left: 0;
    width: 100%;
    z-index: 2;

    background: linear-gradient(to top, rgba(0, 0, 0, 0.8), transparent);
    padding: ${theme.spacings.small}; // Adicionado padding
    color: ${theme.colors.white};
    font-size: ${theme.fonts.sizes.small};

    transform: translateY(100%);
    transition: all 300ms ease-in-out;

    ${GridElement}:hover & {
      opacity: 1;
      transform: translateY(0);
    }
  `}
`;
