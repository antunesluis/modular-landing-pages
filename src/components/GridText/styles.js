import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      opacity: 0.9;
    }

    > ${HeadingContainer} {
      text-shadow: 0.1rem 0.1rem 0.2rem rgba(0, 0, 0, 0.2);
      margin-bottom: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      ${TextComponent} {
        margin-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    max-width: 100%;

    grid-template-columns: repeat(auto-fit, minmax(min(100%, 32rem), 1fr));
    gap: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    transition: all 0.3s ease-in-out;
    border-radius: 10px;

    padding: ${theme.spacings.medium};
    background: rgba(255, 255, 255, 0.03);
    border-radius: ${theme.spacings.tiny};

    &:hover {
      transform: translateY(-6px);
      background: rgba(255, 255, 255, 0.05);
    }

    ${TextComponent} {
      opacity: 0.85;
      line-height: 1.6;
    }
  `}
`;

export const HeaderContainer = styled.div`
  ${() => css`
    ${({ theme }) => css`
      display: flex;
      align-items: center;
      margin-bottom: ${theme.spacings.medium};
      position: relative;

      ${HeadingContainer} {
        margin: 0;
        padding-left: ${theme.spacings.medium};
      }
    `}
  `}
`;

export const CardNumber = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    font-weight: 700;
    // font-weight: bold;
    // font-size: clamp(5rem, 8vw, 7.5rem);

    opacity: 0.15;
    transition: all 0.3s ease-in-out;

    ${GridElement}:hover & {
      opacity: 0.2;
      transform: scale(1.1) rotate(5deg);
    }
  `}
`;
