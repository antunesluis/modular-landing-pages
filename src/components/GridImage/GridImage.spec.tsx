import { renderTheme } from '../../styles/render-theme';
import { GridImage } from '.';
import { screen } from '@testing-library/react';
import mock from './mock';

describe('<GridImage />', () => {
  it('should render with background', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without background', () => {
    const { container } = renderTheme(
      <GridImage {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render image with large as source', () => {
    const customMock = {
      ...mock,
      grid: [
        {
          altText: 'Test',
          large: 'large.jpg',
          medium: 'medium.jpg',
          small: 'small.jpg',
          thumbnail: 'thumb.jpg',
        },
      ],
    };
    renderTheme(<GridImage {...customMock} />);
    expect(screen.getByRole('img', { name: 'Test' })).toHaveAttribute(
      'src',
      'large.jpg',
    );
  });

  it('should render image with medium as source when large does not exist', () => {
    const customMock = {
      ...mock,
      grid: [
        {
          altText: 'Test',
          medium: 'medium.jpg',
          small: 'small.jpg',
          thumbnail: 'thumb.jpg',
        },
      ],
    };
    renderTheme(<GridImage {...customMock} />);
    expect(screen.getByRole('img', { name: 'Test' })).toHaveAttribute(
      'src',
      'medium.jpg',
    );
  });

  it('should render image with small as source when large and medium do not exist', () => {
    const customMock = {
      ...mock,
      grid: [
        {
          altText: 'Test',
          small: 'small.jpg',
          thumbnail: 'thumb.jpg',
        },
      ],
    };
    renderTheme(<GridImage {...customMock} />);
    expect(screen.getByRole('img', { name: 'Test' })).toHaveAttribute(
      'src',
      'small.jpg',
    );
  });

  it('should render image with thumbnail as source when other options do not exist', () => {
    const customMock = {
      ...mock,
      grid: [
        {
          altText: 'Test',
          thumbnail: 'thumb.jpg',
        },
      ],
    };
    renderTheme(<GridImage {...customMock} />);
    expect(screen.getByRole('img', { name: 'Test' })).toHaveAttribute(
      'src',
      'thumb.jpg',
    );
  });

  it('should render altText correctly in image and caption', () => {
    const customMock = {
      ...mock,
      grid: [
        {
          altText: 'Test alternative text',
          thumbnail: 'thumb.jpg',
        },
      ],
    };
    renderTheme(<GridImage {...customMock} />);
    expect(screen.getByRole('img')).toHaveAttribute(
      'alt',
      'Test alternative text',
    );
    expect(screen.getByText('Test alternative text')).toBeInTheDocument();
  });

  it('should render image with srcImg as source when other options do not exist', () => {
    const customMock = {
      ...mock,
      grid: [
        {
          altText: 'Test Source Image',
          srcImg: 'source.jpg',
        },
      ],
    };
    renderTheme(<GridImage {...customMock} />);
    expect(
      screen.getByRole('img', { name: 'Test Source Image' }),
    ).toHaveAttribute('src', 'source.jpg');
  });
});
