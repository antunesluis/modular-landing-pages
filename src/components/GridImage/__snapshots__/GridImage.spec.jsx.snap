// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridImage /> > should render with background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  position: relative;
}

.c2 >.c4 {
  opacity: 0.9;
  max-width: 100%;
}

.c6 {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 3.2rem;
  margin: 3.2rem auto;
}

.c8 {
  aspect-ratio: 1;
  overflow: hidden;
  position: relative;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  border-radius: 1.6rem;
  background: #DDDDDD;
}

.c8::before {
  content: '';
  z-index: 1;
  opacity: 0;
  position: absolute;
  inset: 0;
  background: rgba(0, 0, 0, 0.3);
  transition: opacity 300ms ease-in-out;
}

.c8:hover::before {
  opacity: 1;
}

.c8:hover img {
  transform: scale(1.1) rotate(3deg);
}

.c9 {
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
  vertical-align: middle;
  transition: transform 500ms ease-in-out;
  will-change: transform;
}

.c10 {
  position: absolute;
  opacity: 0;
  bottom: 0;
  right: 0;
  left: 0;
  width: 100%;
  z-index: 2;
  background: linear-gradient(to top, rgba(0, 0, 0, 0.8), transparent);
  padding: 1.6rem;
  color: #FFFFFF;
  font-size: 1.6rem;
  transform: translateY(100%);
  transition: all 300ms ease-in-out;
}

.c7:hover .c10 {
  opacity: 1;
  transform: translateY(0);
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    padding: 3.2rem 1.6rem;
  }

  .c2 >.c4 {
    margin: 2.4rem 0 6.4rem;
  }
}

@media (max-width: 768px) {
  .c6 {
    grid-template-columns: 1fr;
    gap: 2.4rem;
    margin: 2.4rem auto;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Project Gallery
        </h2>
        <div
          class="c4 c5"
        >
          Discover some of the transformative projects we have developed for our clients. Each solution is unique and designed to maximize results.
        </div>
        <div
          class="c6"
        >
          <div
            class="c7 c8"
          >
            <img
              alt="E-commerce Platform"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=1"
            />
            <figcaption
              class="c10"
            >
              E-commerce Platform
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Banking App Interface"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=2"
            />
            <figcaption
              class="c10"
            >
              Banking App Interface
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Healthcare Dashboard"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=3"
            />
            <figcaption
              class="c10"
            >
              Healthcare Dashboard
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Education Platform"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=4"
            />
            <figcaption
              class="c10"
            >
              Education Platform
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Real Estate Solution"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=5"
            />
            <figcaption
              class="c10"
            >
              Real Estate Solution
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Logistics System"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=6"
            />
            <figcaption
              class="c10"
            >
              Logistics System
            </figcaption>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage /> > should render without background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  position: relative;
}

.c2 >.c4 {
  opacity: 0.9;
  max-width: 100%;
}

.c6 {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 3.2rem;
  margin: 3.2rem auto;
}

.c8 {
  aspect-ratio: 1;
  overflow: hidden;
  position: relative;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  border-radius: 1.6rem;
  background: #DDDDDD;
}

.c8::before {
  content: '';
  z-index: 1;
  opacity: 0;
  position: absolute;
  inset: 0;
  background: rgba(0, 0, 0, 0.3);
  transition: opacity 300ms ease-in-out;
}

.c8:hover::before {
  opacity: 1;
}

.c8:hover img {
  transform: scale(1.1) rotate(3deg);
}

.c9 {
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
  vertical-align: middle;
  transition: transform 500ms ease-in-out;
  will-change: transform;
}

.c10 {
  position: absolute;
  opacity: 0;
  bottom: 0;
  right: 0;
  left: 0;
  width: 100%;
  z-index: 2;
  background: linear-gradient(to top, rgba(0, 0, 0, 0.8), transparent);
  padding: 1.6rem;
  color: #FFFFFF;
  font-size: 1.6rem;
  transform: translateY(100%);
  transition: all 300ms ease-in-out;
}

.c7:hover .c10 {
  opacity: 1;
  transform: translateY(0);
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    padding: 3.2rem 1.6rem;
  }

  .c2 >.c4 {
    margin: 2.4rem 0 6.4rem;
  }
}

@media (max-width: 768px) {
  .c6 {
    grid-template-columns: 1fr;
    gap: 2.4rem;
    margin: 2.4rem auto;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Project Gallery
        </h2>
        <div
          class="c4 c5"
        >
          Discover some of the transformative projects we have developed for our clients. Each solution is unique and designed to maximize results.
        </div>
        <div
          class="c6"
        >
          <div
            class="c7 c8"
          >
            <img
              alt="E-commerce Platform"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=1"
            />
            <figcaption
              class="c10"
            >
              E-commerce Platform
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Banking App Interface"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=2"
            />
            <figcaption
              class="c10"
            >
              Banking App Interface
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Healthcare Dashboard"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=3"
            />
            <figcaption
              class="c10"
            >
              Healthcare Dashboard
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Education Platform"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=4"
            />
            <figcaption
              class="c10"
            >
              Education Platform
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Real Estate Solution"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=5"
            />
            <figcaption
              class="c10"
            >
              Real Estate Solution
            </figcaption>
          </div>
          <div
            class="c7 c8"
          >
            <img
              alt="Logistics System"
              class="c9"
              loading="lazy"
              src="https://picsum.photos/360/360?r=6"
            />
            <figcaption
              class="c10"
            >
              Logistics System
            </figcaption>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
